ocs/volumes.py

```diff
@@ -0,0 +1,257 @@
+                                  Recycle or Delete
+            fstype (str): filesystem type
+            clusterNamespace (str): namespace where rook cluster exists
+
+        Returns:
+            str: name of the storage class created
+
+        Example:
+            create_storageclass(
+                blockPool,
+                sc_name_prefix="autotests-sc",
+                allow_volume_expansion=True,
+                reclaim_policy="Delete",
+                fstype="xfs"
+                clusternamespace="openshift-storage",
+            )
+
+        """
+        if self.name:
+            sc_name = self.name
+        else:
+            sc_name = f"{sc_name_prefix}-{get_random_str()}"
+
+        sc_data = {}
+        sc_data['k8s_api_version'] = defaults.STORAGE_API_VERSION
+        sc_data['storageclass_name'] = sc_name
+        sc_data['volume_expansion'] = allow_volume_expansion
+        sc_data['reclaimPolicy'] = reclaim_policy
+        sc_data['blockPool'] = blockPool
+        sc_data['clusterNamespace'] = clusterNamespace
+        sc_data['fstype'] = fstype
+
+        data = generate_yaml_from_jinja2_template_with_data(
+            self.template_path,
+            **sc_data
+        )
+        self.service_sc.create(body=data)
+
+        return sc_name
+
+
+class PVC(KubeVolume):
+    """
+    Class which contains PVC related functionality
+
+    Attributes:
+        name (str): name of the PVC volume
+        namespace (str): namespace to create PVC
+
+    """
+    def __init__(self, name=None, namespace='default'):
+        KubeVolume.__init__(self, name, namespace)
+        self.kind = kinds.PVC
+        self.api_version = defaults.OPENSHIFT_REST_CLIENT_API_VERSION
+        self.service_pvc = self.dyn_client.resources.get(
+            api_version=self.api_version,
+            kind=self.kind
+        )
+        self.template_path = os.path.join(TEMPLATE_DIR, "pvc.yaml")
+
+    def create_pvc(
+        self,
+        storageclass,
+        accessmode="ReadWriteOnce",
+        pvc_name_prefix="autotests-pvc",
+        pvc_size=3
+    ):
+        """
+        Creates PVC using data provided
+
+        Args:
+            storageclass (str): name of storageclass to create PVC
+            accessmode (str): access mode for PVC
+            pvc_name_prefix (str): prefix given to PVC name
+            pvc_size (int): size of PVC in Gb
+
+        Returns:
+            str: name of the pvc created
+
+        Examples:
+            create_pvc(
+                storageclass,
+                accessmode="ReadWriteOnce",
+                pvc_size=3
+            )
+            create_pvc(
+                storageclass,
+                accessmode="ReadWriteOnce ReadOnlyMany",
+                pvc_size=5
+            )
+
+        """
+        if self.name:
+            pvc_name = self.name
+        else:
+            pvc_name = f"{pvc_name_prefix}-{get_random_str()}"
+        pvc_size = f"{pvc_size}Gi"
```

Comment:
That's really good!
I'm thinking if we can move this functionality directly to the jinja2 yaml.
something like - {{ size"Gi" | default(3Gi) }}
Not sure if that'll work

Reply:
Used the same in yaml. Above variable gives the flexibility for user to define their own pvc size